nifpga/session.py

```diff
@@ -390,8 +410,272 @@ def read(self):
+        super(_FxpRegister, self).__init__(session,
+                                           nifpga,
+                                           bitfile_register,
+                                           base_address_on_device)
+
+        self._word_length = bitfile_register.word_length
+        self._integer_word_length = bitfile_register.integer_word_length
+        self._radix_point = self._calculate_radix_point()
+        self._overflow_enabled = bitfile_register.overflow
+        self._signed = bitfile_register.signed
+        self._delta = self._calculate_delta(bitfile_register.word_length,
+                                            bitfile_register.integer_word_length)
+
+    @property
+    def overflow(self):
+        """ This property should only set be whenever this register has overflow
+        enabled"""
+        return self._overflow
+
+    @overflow.setter
+    def overflow(self, value):
+        if self._overflow_enabled:
+            self._overflow = value
+
+    def _calculate_delta(self, word_length, integer_word_length):
+        return 2**(integer_word_length - word_length)
+
+    def read(self):
+        """ Reads the entire FXP number as an array U32 bits then combining
+        the array bits into a single datatype, then convert it into python
+        Decimal. The largest expecting transfer would be a 64 bit fixed point
+        with the overflow status enabled (making an array of length 3).
+
+        Returns:
+            data (DataType.value): The data in the register in a python list.
+        """
+        data = super(_FxpRegister, self).read()
+        data = self._combine_array_of_u32_into_one_value(data)
+        return self._convert_from_read_value_to_decimal(data)
+
+    def _combine_array_of_u32_into_one_value(self, data):
+        combinedData = 0
+        for index in range(0, len(self)):
+            combinedData = (combinedData << 32 * index) + data[index]
+        if len(self) > 1:
+            overflow_bit = 1 if self._overflow_enabled else 0
+            total_num_bits = self._word_length + overflow_bit
+            combinedData = combinedData >> (32 * len(self) - total_num_bits)
+        return combinedData
+
+    def _convert_from_read_value_to_decimal(self, data):
+        """ This function converts read value from a RIO device and sets
+        the overflow bit if enabled and returns a Decimal representation of
+        the fixed point. """
+        if self._overflow_enabled:
+            self.overflow = self._get_overflow_value(data)
+            data = self._remove_overflow_bit(data)
+        if self._signed:
+            data = self._integer_twos_comp(data)
+        return Decimal(data * self._delta)
+
+    def _get_overflow_value(self, data):
+        """ Mask out all the data within the word length, leaving the overflow
+        bit. If the result after masking the the word portion of the fixed
+        point is none zero that indicates the data read has overflown."""
```

Comment:
overflown? Is that present perfect progressive tense, like "Baby, my heart is overflow'n"? Or maybe overflowed.